src/Core/Mesh/TriangleMesh.hpp

```diff
@@ -19,67 +17,83 @@ namespace Core
 /// The VertexAttribManager allows to dynammicaly add VertexAttrib per Vertex
 /// See MeshUtils for geometric functions operating on a mesh.
 /// Points and Normals are always present, accessible with
-/// points() and normals()
+/// points() and normals().
 /// Other attribs could be added with attribManager().addAttrib() and
 /// accesssed with attribManager().getAttrib()
-class TriangleMesh
-{
+/// Note: attribute names "in_position" and "in_normal" are reserved.
+class TriangleMesh {
   public:
     EIGEN_MAKE_ALIGNED_OPERATOR_NEW
 
-    using PointAttribHandle  = AttribHandle<Vector3>;
+    using PointAttribHandle = AttribHandle<Vector3>;
     using NormalAttribHandle = AttribHandle<Vector3>;
-    using FloatAttribHandle  = AttribHandle<float>;
-    using Vec2AttribHandle   = AttribHandle<Vector2>;
-    using Vec3AttribHandle   = AttribHandle<Vector3>;
-    using Vec4AttribHandle   = AttribHandle<Vector4>;
-    using Face               = VectorNui;
+    using FloatAttribHandle = AttribHandle<float>;
+    using Vec2AttribHandle = AttribHandle<Vector2>;
+    using Vec3AttribHandle = AttribHandle<Vector3>;
+    using Vec4AttribHandle = AttribHandle<Vector4>;
+    using Face = VectorNui;
 
     /// Create an empty mesh.
     inline TriangleMesh() { initDefaultAttribs(); }
-    /// Copy constructor and assignment operator
-    /// \warning No deep copy of attributes !
+
+    /// Copy constructor and assignment operator.
+    /// Note: shallow copy of attributes.
     TriangleMesh( const TriangleMesh& ) = default;
     TriangleMesh& operator=( const TriangleMesh& ) = default;
 
+    /// Copy only the required attributes (deep copy).
+    /// Note: Passing directly through the AttribManager is highly discouraged.
+    template <typename... Handles>
+    void copyAttribs( const TriangleMesh& input, Handles... attribs );
+
     /// Erases all data, making the mesh empty.
+    /// Note: invalidates shallow copies.
     inline void clear();
 
     /// Appends another mesh to this one.
     /// \todo handle attrib here as well !
     inline void append( const TriangleMesh& other );
 
-    VectorArray<Triangle> m_triangles;
-    VectorArray<Face> m_faces;
-
-    PointAttribHandle::Container& vertices()
-    {
+    /// Access the vertices positions.
+    PointAttribHandle::Container& vertices() {
         return m_vertexAttribs.getAttrib( m_verticesHandle ).data();
     }
-    NormalAttribHandle::Container& normals()
-    {
-        return m_vertexAttribs.getAttrib( m_normalsHandle ).data();
+    const PointAttribHandle::Container& vertices() const {
+        return m_vertexAttribs.getAttrib( m_verticesHandle ).data();
     }
 
-    const PointAttribHandle::Container& vertices() const
```

Comment:
Why removing the `vertices () const` method ?

Reply:
actually it hasn't been removed, the style added the `{` at the end of the line

Reply:
ok, my bad, the diff visualization was misleading